### PR TITLE
Walljump changes

### PR DESCRIPTION
**Pull request by JXVD**

### CODE REVIEW

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 1]
```diff
@@ -4,51 +4,60 @@
...
+
+    [SerializeField]
     private PlayRandomSource wallSlideSound;
     [SerializeField]
     private PlayRandomSource jumpSound;
 
     private float timeToWallUnstick;
 
     private bool jumped = false;
+    protected override void Start()
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: Why is this here?

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 2]
```diff
@@ -57,32 +66,45 @@ public override Vector3 ApplyFriction(Vector3 velocity)
...
             {
-                outVelocity.x = -wallDirX * wallJumpClimbForces.x;
-                outVelocity.y = wallJumpClimbForces.y;
+                outVelocity.x = (-wallDirX) * (wallJumpVelocity.x / 2);
+                outVelocity.y = wallJumpVelocity.y;
             }
-            //if the player has no x input
-            else if (moveX == 0)
+            //if the input is down and slightly away
+            else if(Mathf.Abs(wallDirX - direction.x) < 1.16  && Mathf.Abs(1 - direction.y) > .25)
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: I really don't like having magic numbers like this. These should be moved out into variables that describe what they actually are.

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 3]
```diff
@@ -4,51 +4,60 @@
...
+    }
+
 
     public override Vector3 ApplyFriction(Vector3 velocity)
     {
         int wallDirX = (controller.collisions.Left) ? -1 : 1;
         float moveX = player.InputPlayer.GetAxis("Move Horizontal");
-        velocity.x = moveX;
-        if (timeToWallUnstick > 0)
+        if (!jumped)
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: nit: You might want to do the inverse of this check, and just return velocity instead. Then you can un-nest the next if statement

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 4]
```diff
@@ -57,32 +66,45 @@ public override Vector3 ApplyFriction(Vector3 velocity)
     {
         base.ApplyInputs(inVelocity, inExternalForces, out outVelocity, out outExternalForces);
 
+        //normalize the direction so that it just gives the wall jump angle
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: nit: this isn't an angle

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 5]
```diff
@@ -57,32 +66,45 @@ public override Vector3 ApplyFriction(Vector3 velocity)
...
+        direction = player.AimDirection.normalized;
         int wallDirX = (controller.collisions.Left) ? -1 : 1;
-        float moveX = player.InputPlayer.GetAxis("Move Horizontal");
 
         jumped = false;
+
+        Debug.Log("direction x: " + direction.x + " direction y: " + direction.y);
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: This is gonna clutter the logs, do we need it here?

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 6]
```diff
@@ -57,32 +66,45 @@ public override Vector3 ApplyFriction(Vector3 velocity)
...
+                outVelocity.x = -wallDirX * wallJumpVelocity.x * 2;
+                outVelocity.y = wallJumpVelocity.y;
+
+            }
+            //if the input is pointed towards the wall or slightly away and up
+            else if (Mathf.Abs(wallDirX - direction.x) < 0.5f ||  ((Mathf.Abs(wallDirX - direction.x) < 1.16) && (Mathf.Abs(1 - direction.y) < .25 )))
             {
-                outVelocity.x = -wallDirX * wallJumpClimbForces.x;
-                outVelocity.y = wallJumpClimbForces.y;
+                outVelocity.x = (-wallDirX) * (wallJumpVelocity.x / 2);
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: this would already be done by 82

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 7]
```diff
@@ -57,32 +66,45 @@ public override Vector3 ApplyFriction(Vector3 velocity)
...
+                outVelocity.y = wallJumpVelocity.y;
             }
-            //if the player has no x input
-            else if (moveX == 0)
+            //if the input is down and slightly away
+            else if(Mathf.Abs(wallDirX - direction.x) < 1.16  && Mathf.Abs(1 - direction.y) > .25)
             {
-                outVelocity.x = -wallDirX * wallJumpOffForces.x;
-                outVelocity.y = wallJumpOffForces.y;
+                outVelocity.x = (-wallDirX) * (wallJumpVelocity.x / 2);
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: also done by 82

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 8]
```diff
@@ -4,51 +4,60 @@
...
-    private Vector2 wallJumpClimbForces;
-    [SerializeField]
-    private Vector2 wallJumpOffForces;
-    [SerializeField]
-    private Vector2 wallLeapForces;
-    [SerializeField]
     private float wallStickTime;
 
     [SerializeField]
+    private Vector2 wallJumpVelocity;
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: Overall I see why we're doing the configuration like this, but I'm not sure this is going to be enough to let us configure it as we need. We might want to pull out the modifiers for each scenario into accessible variables.

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs [hunk 9]
```diff
@@ -4,51 +4,60 @@
...
+
+
+    private Vector2 direction;
+
+    //Wallride variables being made accessable
+
+    public float WallStickTime { get { return wallStickTime; } }
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: are we planning to consume this or are we exposing it just in case?

**Review by Juskelis — CHANGES_REQUESTED** [review]

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs (hunk 5 above)
Comment (JXVD) [reply Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: no, left it in by accident

File: Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs (hunk 2 above)
Comment (JXVD) [reply Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs]: Yes, this is true

**Review by Juskelis — APPROVED** [review]
